virtual-list-element.js

```diff
@@ -63,12 +63,14 @@ export class VirtualListElement extends HTMLElement {
   }
 
   static get observedAttributes() {
-    return ['layout'];
+    return ['layout', 'items'];
   }
 
   attributeChangedCallback(name, oldVal, newVal) {
     if (name === 'layout') {
       this.layout = newVal;
+    } else if (name === 'items') {
```

Comment:
Don't do this pattern for reflecting attributes. Instead, define

```js
get size() {
  return +this.getAttribute("size");
}

set size(v) {
  this.setAttribute("size", +v);
}
```

and then use `this.size` internally. This handles several edge cases better.

Reply:
For `size` we don't require to reflect it to attribute. We do that for `layout` because we have need it for styling.
Also, `size` must be a number, so it's more performant to do the conversion only in the `attributeChangedCallback` and expect users to pass in a number when using the setter.

Reply:
It's not really OK to have a property that sometimes picks up its value from an attribute, but doesn't reflect back. That's not something we see in native DOM elements. They all use the pattern I showed above.

In general you cannot expect users to pass in the right types in DOM APIs. They all do conversions.

Reply:
Ok, made all reflecting and always reading from the attribute now